src/backend/tcop/postgres.c

```diff
@@ -4592,7 +4592,10 @@ PostgresMain(const char *dbname, const char *username)
 			ReportChangedGUCOptions();
 
 			if (MyProcPort && MyProcPort->protocol_config->fn_send_ready_for_query)
+			{
+				Assert(MyProcPort != NULL);
```

Comment:
Unnecessary assert here as we are already checking if `MyProcPort` is not null in the condition above.
Additionally, we also need to add a check for `MyProcPort` in `ReadCommand` isn't it?

Reply:
Right, got it confused with something else. Updated in latest rev